api/v1alpha3/azuremachine_types.go

```diff
@@ -44,6 +58,11 @@ type AzureMachineSpec struct {
 	// +optional
 	Image *Image `json:"image,omitempty"`
 
+	// IdentityType is the type of identity used for the virtual machine scale set.
+	// The type 'SystemAssigned, UserAssigned' includes both an implicitly created identity
+	// and a set of user assigned identities.
+	IdentityType MachineIdentityType `json:"identityType,omitempty"`
```

Comment:
I noticed there was a type `VMIdentity` in types.go, not sure if it was ever used but it doesn't look like it's being used currently, I think we should remove it to avoid confusion

Reply:
or should I switch to using `VMIdentity`? do you have a preference?

Reply:
Actually, I have a slight preference for `VMIdentity` because it makes it clear that this is an Azure VM property and not a cluster API "machine" property. What do you and @devigned think?

Reply:
yeah, makes sense, I can switch to that, to be honest I didn't see it

Reply:
What about AzureMachinePools? They will use the same identity behaviors. Does VMIdentity still sound good when using it for AzureMachinePools?

Reply:
I think we need to solve this problem in general, not just for identity. What about VMSize? Right now we use `VM` in the codebase to mean both "an Azure virtual machine resource" and "an Azure compute unit", we should probably decide on one. In "Azure" talk, an individual VMSS compute unit is called an "instance" but then VMSS also stands for "Virtual Machine Scale Set" so it's confusing...

Reply:
FWIW the official Azure docs call VMSS a "group of VMs", so I think it's okay to say VM is either a standalone VM or a VM part of a scale set.

https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/overview

Reply:
makes sense to me to use the VMIdentity